python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+
+        fieldNames = []
+        fieldRAs = []
+        fieldDecs = []
+        epochs = []
+        fieldRegions = {}
+
+        for i in range(clusters.n_clusters_):
+            fieldInd = clusters.labels_ == i
+            fieldDetectors = sum([allDetectorCorners[f] for f, fInd in enumerate(fieldInd) if fInd], [])
+            hull = lsst.sphgeom.ConvexPolygon.convexHull(fieldDetectors)
+            center = lsst.geom.SpherePoint(hull.getCentroid())
+            ra = center.getRa().asDegrees()
+            dec = center.getDec().asDegrees()
+
+            fieldRegions[i] = hull
+            fieldNames.append(str(i))
+            fieldRAs.append(ra)
+            fieldDecs.append(dec)
+            epochs.append(medianEpoch)
+
+        fields = wcsfit.Fields(fieldNames, fieldRAs, fieldDecs, epochs)
+
+        return fields, fieldRegions
+
+    def _associate_from_isolated_sources(
+        self, isolatedStarSourceRefs, isolatedStarCatalogRefs, extensionInfo, refObjects
+    ):
+        """Match the input catalog of isolated stars with the reference catalog
+        and transform the combined isolated star sources and reference source
+        into the format needed for gbdes.
+
+        Parameters
+        ----------
+        isolatedStarSourceRefs : `list`
+            List of `DeferredDatasetHandle`s pointing to isolated star sources.
+        isolatedStarCatalogRefs: `list`
+            List of `DeferredDatasetHandle`s pointing to isolated star
+            catalogs.
+        extensionInfo : `lsst.pipe.base.Struct`
+            Struct containing properties for each extension.
+        refObjects : `dict`
+            Dictionary of dictionaries containing the position and error
+            information of reference objects.
+
+        Returns
+        -------
+        associations : `lsst.pipe.base.Struct`
+            Struct containing the source associations.
+        sourceDict : `dict`
+            Dictionary containing the source centroids for each visit.
+        """
+        sequences = []
+        extensions = []
+        object_indices = []
+
+        sourceColumns = ["x", "y", "xErr", "yErr", "ixx", "ixy", "iyy", "obj_index", "visit", "detector"]
+        catalogColumns = ["ra", "dec"]
+
+        sourceDict = dict([(visit, {}) for visit in np.unique(extensionInfo.visit)])
+        for visit, detector in zip(extensionInfo.visit, extensionInfo.detector):
+            sourceDict[visit][detector] = {"x": [], "y": [], "xCov": [], "yCov": [], "xyCov": []}
+
+        for isolatedStarCatalogRef, isolatedStarSourceRef in zip(
+            isolatedStarCatalogRefs, isolatedStarSourceRefs
+        ):
+            isolatedStarCatalog = isolatedStarCatalogRef.get(parameters={"columns": catalogColumns})
+            isolatedStarSources = isolatedStarSourceRef.get(parameters={"columns": sourceColumns})
+            if len(isolatedStarCatalog) == 0:
+                continue
```

Comment:
This is expected to happen for tracts on the edge of an observed region, where maybe only a single visit overlaps with the tract. It will happen a lot and isn't really informative unless something upstream went really haywire, so debug is probably best.